caddyhttp/basicauth/basicauth_test.go

```diff
@@ -14,19 +14,31 @@ import (
 )
 
 func TestBasicAuth(t *testing.T) {
+	var i int
+	// This handler is registered for tests in which the only authorized user is
+	// "okuser"
+	upstreamHandler := func(w http.ResponseWriter, r *http.Request) (int, error) {
+		remoteUser, _ := r.Context().Value("remote_user").(string)
```

Comment:
Admittedly, it could be _too_ clever. Maybe something like [this](https://github.com/mholt/caddy/blob/30de2e47590bdd63219b259c87b281e0f5ec00d1/caddyhttp/httpserver/context.go#L327-L332) is clearer for maintenance. Does it make sense to have some helper functions for setting and getting context variables?